src/vanilla/Templates/Rest/Common/EnumTemplate.cshtml

```diff
@@ -9,7 +9,52 @@ namespace @(Settings.Namespace).@(Settings.ModelsName)
     /// <summary>
     @WrapComment("/// ", "Defines values for " + Model.Name + ".")
     /// </summary>
-    @if (!Model.ModelAsString)
+    @if(Settings.CustomSettings.ContainsKey("ExtensibleEnums") && (bool)Settings.CustomSettings["ExtensibleEnums"])
+    {
+    var underlyingType = (Model.UnderlyingType == null)? "string" : Model.UnderlyingType.ToString();
```

Comment:
@dsgouda - The fix in this [PR]((https://github.com/Azure/autorest.modeler/pull/22/files#diff-1f2c985b9819bf215e2376529fb1ba78R63)) should help you with underlyingType.

Reply:
Yup, I realized that, and I am kinda waiting on it.

Reply:
It is merged. You should be good to go with a dependency on newer version of the modeler